### PR TITLE
CRYP-100: welcome screen

### PR DESCRIPTION
**Pull request by jason-gerard**
JIRA: https://cryptify.atlassian.net/browse/CRYP-100

## Changes
- Add welcome screen when user is signed out
- Implement signed in and signed out flow
- Add back buttons on sign in and sign up screens
- Expose jwt token as context state to all screens

### CODE REVIEW

File: packages/client/screens/WelcomeScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+export default function WelcomeScreen({ navigation }: GuestStackScreenProps<"WelcomeScreen">) {
+    return (
+        <View style={styles.view}>
+            <Box safeArea></Box>
+            <Box style={styles.titleContainer}>
+                <Logo />
+                <Box marginTop="20px"></Box>
+                <Text style={styles.subtext}>Crypto asset management{"\n"}made simple</Text>
+            </Box>
```
Comment (domsec) [inline packages/client/screens/WelcomeScreen.tsx:17]: Wouldn't it be better to refactor this with `VStack` similar to what we did in Sign In/Sign Up screens?

File: packages/client/screens/WelcomeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            <Button onPress={() => navigation.navigate("SignUpScreen")}>Sign up</Button>
+            <Box marginTop="15px"></Box>
+            <Button variant="outline" onPress={() => navigation.navigate("SignInScreen")}>
+                Sign in
+            </Button>
+            <Box marginBottom="20px"></Box>
```
Comment (domsec) [inline packages/client/screens/WelcomeScreen.tsx:23]: Same as above comment.

**Review by domsec — APPROVED** [review]
Besides my two refactor comments, everything looks good. I'm really glad the welcome screen background alignment is a 1 to 1 match with the mockups.

File: packages/client/screens/WelcomeScreen.tsx (hunk 1 above)
Comment (jason-gerard) [reply packages/client/screens/WelcomeScreen.tsx]: yep you are right, updated

File: packages/client/screens/WelcomeScreen.tsx (hunk 2 above)
Comment (jason-gerard) [reply packages/client/screens/WelcomeScreen.tsx]: updated

**Review by domsec — APPROVED** [review]
Great work.

**Pull request merged**